python/lsst/meas/algorithms/makePsfCandidates.py

```diff
@@ -0,0 +1,141 @@
+#
+# LSST Data Management System
+#
+# Copyright 2008-2017  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+__all__ = ["MakePsfCandidatesConfig", "MakePsfCandidatesTask"]
+
+import numpy as np
+
+from lsst.afw.table import SourceCatalog, Schema
+import lsst.afw.math as afwMath
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from . import makePsfCandidate
+from future.utils import with_metaclass
+
+
+class MakePsfCandidatesConfig(pexConfig.Config):
+    kernelSize = pexConfig.Field(
+        doc="size of the kernel to create",
+        dtype=int,
+        default=21,
+    )
+    borderWidth = pexConfig.Field(
+        doc="number of pixels to ignore around the edge of PSF candidate postage stamps",
+        dtype=int,
+        default=0,
+    )
+
+
+class MakePsfCandidatesTask(pipeBase.Task):
+    """Create PSF candiates given an input catalog.
+    """
+    ConfigClass = MakePsfCandidatesConfig
+    _DefaultName = "makePsfCandidates"
+
+    def __init__(self, **kwds):
+        pipeBase.Task.__init__(self, **kwds)
+
+    def run(self, starCat, exposure=None, isStarField=None):
+        """Make a list of PSF candidates from a star catalog.
+
+        Parameters
+        ----------
+        starCat : `lsst.afw.table.SourceCatalog`
+            Catalog of stars, as returned by
+            ``lsst.meas.algorithms.starSelector.run()``.
+        exposure : `lsst.afw.image.Exposure` (optional)
+            The exposure containing the sources.
+
```

Comment:
Good point. I've renamed it `starSelectedField`, since it's actually `str`, not a `bool`.

This is, however, reminding me that we don't have a unittest for makePsfCandidates...

Reply:
Great idea! How about a unit test! 😸

Reply:
Based on my work on the new test, I've renamed it again to `psfCandidateField`, which I think is even better. And there's a new unittest in another commit.